pkg/controller/externalsecrets_controller.go

```diff
@@ -159,11 +166,14 @@ func BuildCustomClient(mgr ctrl.Manager, r *ExternalSecretsReconciler) (client.C
 			return nil, fmt.Errorf("failed to add informer for %s resource: %w", certificateObject.GetObjectKind().GroupVersionKind().String(), err)
 		}
 	}
-	ownObject := &operatorv1alpha1.ExternalSecrets{}
 	_, err = customCache.GetInformer(context.Background(), ownObject)
 	if err != nil {
 		return nil, fmt.Errorf("failed to add informer for %s resource: %w", ownObject.GetObjectKind().GroupVersionKind().String(), err)
 	}
+	_, err = customCache.GetInformer(context.Background(), esmObject)
```

Comment:
looks buggy to me,

Seems like `controllerManagedResources` above lacks the `&operatorv1alpha1.ExternalSecretsManager{}` object. Essentially, 

```
for _, res := range controllerManagedResources {
		switch res {
                    // ...
		     ...Watches(res, handler.EnqueueRequestsFromMapFunc(mapFunc), managedResourcePredicate)
		}
	}
```
will ultimatley not call `.Watches(&operatorv1alpha1.ExternalSecretsManager{})` in SetupWithManager.

This might lead to the cache not getting updating making the related informer not useful, I don't think this is intended? Instead you might just add ESM into the watched resource list and work fine (/ alternate is to run cache without client, direct apiserver calls).

Reply:
It's added to the watch [here](https://github.com/openshift/external-secrets-operator/blob/2f5f2a8b9e198fd22836608fbae3c05ea1a89e6a/pkg/controller/externalsecrets_controller.go#L233), which was already present. Do have it right, this is one is question correct?

Reply:
suggestion incorporated.